active/4-SSAS-ingestion.md

```diff
@@ -0,0 +1,95 @@
+
+
+## Requirements
+
+- Ingestion metadata from SSAS Tabular models
+- Ingestion metadata from SSAS Multidimensional models
+
+
+### Extensibility
+
+- Build lineage to/from SSAS models
+
+## Detailed design
+
+General information about [OLAP cubes](https://learn.microsoft.com/en-us/system-center/scsm/olap-cubes-overview?view=sc-sm-2022).
+
+
+The interaction with SSAS (SQL Server Analysis Services) is carried out through [Microsoft's solution](https://learn.microsoft.com/en-us/analysis-services/instances/configure-http-access-to-analysis-services-on-iis-8-0?view=asallproducts-allversions).
+
+Arguments in favor of such a solution:
+- Cross-platform compatibility.
+- A single, standardized entry point for working with SSAS.
+
+
+General scheme.
+```mermaid
+graph LR;
+    id1[DataHub]---id2[IIS web server];
+    id2[IIS web server]---id3[SSAS1];
+    id2[IIS web server]---id4[SSAS2];
+```
+Data exchange occurs using XMLA queries wrapped in HTTP.
+- For multidimensional SSAS servers, a [DISCOVER_XML_METADATA](https://learn.microsoft.com/en-us/openspecs/sql_server_protocols/ms-ssas/51647299-75c7-471d-896f-a691e4114b18) type query is used.
+- For tabular SSAS servers, [DMV](https://learn.microsoft.com/en-us/analysis-services/instances/use-dynamic-management-views-dmvs-to-monitor-analysis-services?view=asallproducts-allversions) (Dynamic Management View) queries are utilized.
+
+
+
+The following scheme was proposed for entity mapping:
+```mermaid
+graph TB;
+    c1---b1;
+    b1---a1;
+    b1---a2;
+    subgraph s1[Properties];
+    a1["Dimension"];
+    a2["Measure"];
+    end;
+    subgraph s2[DataSet];
+    b1["Cube"];
+    end;
+    subgraph s3[Container];
+    c1["Catalog(database)"];
+    end;
+```
+- Server maps to a container.
+- Catalog maps to a container (and is hierarchically nested within the server container).
+- Cube is mapped as a dataset.
+- Dimension and measure become properties of the dataset.
+## How we teach this
+
+We should create/update user guides to educate users for:
+ - Search & discovery experience (how to find a SSAS models in DataHub)
+ - Lineage experience (how to find different entities connected to the SSAS models)
+
+## Rollout / Adoption Strategy
+
+If it will be standalone module only who want will use it. So we no need any migration tools and braking changes. 
+
+## Future Work
+
+Establish a complete data lineage from the data source to the analytical models.
+
+## Unresolved questions
+
+- It would be better to create this module as a standalone, focusing solely on SSAS, or should it be integrated into the existing Mssql module?
+- Is it relevant to add SSAS entities (catalog, cube, dimension, measure) to the DataHub?
```

Comment:
The mapping you described above (catalog -> container, etc) should be fine. For dimensions and measures, we can model those as schema fields with tags of "Dimension" or "Measure". We already do something similar for Looker. 

Eventually we want to add dimensions/measures as more first class things within datahub, and can migrate accordingly when the time comes.